diff --git a/docroot/sites/all/modules/contrib/piwik/piwik.module b/docroot/sites/all/modules/contrib/piwik/piwik.module
index 540d9bbf6..9b1632833 100644
--- a/docroot/sites/all/modules/contrib/piwik/piwik.module
+++ b/docroot/sites/all/modules/contrib/piwik/piwik.module
@@ -180,6 +180,15 @@ function piwik_page_alter(&$page) {
       }
     }
 
+    if (variable_get('piwik_translation_set', 0)) {
+      $node = menu_get_object();
+      if ($node && $node->nid) {
+        global $language;
+        $set_custom_url = drupal_json_encode(url('node/' . $node->nid, array('absolute' => TRUE)));
+        $set_custom_url = str_replace('\/' . $language->language . '\/', '\/', $set_custom_url);
+      }
+    }
+
     // Track access denied (403) and file not found (404) pages.
     if ($status == '403 Forbidden') {
       $set_document_title = '"403/URL = " + encodeURIComponent(document.location.pathname+document.location.search) + "/From = " + encodeURIComponent(document.referrer)';
