diff --git a/docroot/sites/all/modules/contrib/piwik/piwik.admin.inc b/docroot/sites/all/modules/contrib/piwik/piwik.admin.inc
index 312d4c8f4..484bebdef 100644
--- a/docroot/sites/all/modules/contrib/piwik/piwik.admin.inc
+++ b/docroot/sites/all/modules/contrib/piwik/piwik.admin.inc
@@ -306,7 +306,7 @@ function piwik_admin_settings_form($form_state) {
   $piwik_custom_vars = variable_get('piwik_custom_var', array());
 
   // Piwik supports up to 5 custom variables.
-  for ($i = 1; $i < 6; $i++) {
+  for ($i = 1; $i < 11; $i++) {
     $form['piwik_custom_var']['slots'][$i]['slot'] = array(
       '#default_value' => $i,
       '#description' => t('Slot number'),
diff --git a/docroot/sites/all/modules/contrib/piwik/piwik.module b/docroot/sites/all/modules/contrib/piwik/piwik.module
index db682abad..540d9bbf6 100644
--- a/docroot/sites/all/modules/contrib/piwik/piwik.module
+++ b/docroot/sites/all/modules/contrib/piwik/piwik.module
@@ -191,7 +191,7 @@ function piwik_page_alter(&$page) {
     // Add custom variables.
     $piwik_custom_vars = variable_get('piwik_custom_var', array());
     $custom_variable = '';
-    for ($i = 1; $i < 6; $i++) {
+    for ($i = 1; $i < 11; $i++) {
       $custom_var_name = !empty($piwik_custom_vars['slots'][$i]['name']) ? $piwik_custom_vars['slots'][$i]['name'] : '';
       if (!empty($custom_var_name)) {
         $custom_var_value = !empty($piwik_custom_vars['slots'][$i]['value']) ? $piwik_custom_vars['slots'][$i]['value'] : '';
