diff --git a/editors/js/tinymce-3.js b/editors/js/tinymce-3.js
index 83bae13..8f1b7cc 100644
--- a/editors/js/tinymce-3.js
+++ b/editors/js/tinymce-3.js
@@ -226,7 +226,16 @@ Drupal.wysiwyg.editor.instance.tinymce = {
         });
       });
     });
-    return $content.html();
+    var html = '';
+    $content.each(function(){
+      if ($(this)[0].localName == 'script') {
+        html += $(this)[0].outerHTML;
+      }
+      else {
+        html += $(this)[0].innerHTML;
+      }
+    });
+    return html;
   },
 
   insert: function(content) {
